lightbeam/send.py

```diff
@@ -109,10 +156,27 @@ async def do_post(self, endpoint, file_name, data, client, line, hash):
                         # warn about errors
                         if response.status not in [ 200, 201 ]:
                             message = str(response.status) + ": " + util.linearize(body)
+
+                            # update run metadata...
+                            if "failed_statuses" not in self.metadata["resources"][endpoint].keys():
```

Comment:
Consider isolating the failed statuses dict as a variable to streamline updates.

```
failed_statuses_dict = self.metadata["resources"][endpoint].get("failed_statuses", {})

if response.status not in failed_statuses_dict:
    failed_statuses_dict.update({response.status: {}})

if message not in failed_statuses_dict[response.status]:
    failed_statuses_dict[response.status].update({message: {}})

if "files" not in failed_statuses_dict[response.status][message]:
    failed_status_dict[response.status][message].update({"files": {}})

if file_name not in failed_statuses_dict[response.status][message]["files"]:
    failed_statuses_dict[response.status][message]["files"].update({file_name: {}})

if "line_numbers" not in failed_statuses_dict[response.status][message]["files"][file_name]:
    failed_statuses_dict[response.status][message]["files"][file_name].update({"line_numbers": []})

failed_statuses_dict[response.status][message]["files"][file_name]["line_numbers"].append(line)
```

Additionally, this is a very example of a defaultdict use-case, but it's multiple layers nested which might be difficult to set-up. If we were able to do so, it would convert this entire code-block to the following:
```
self.metadata["resources"][endpoint]["failed_statuses"][response.status][message]["files"][file_name]["line_numbers"].append(line)
```
I'm going to play around with this concept a little more.

Reply:
@jayckaiser I made another update along these lines in [this commit](https://github.com/edanalytics/lightbeam/pull/10/commits/4c85b22dac6fa4692f8c85807c47eb5a24c7ef3a) and I tested it. If this is good enough for now, I can merge and we could consider refining later?